spec/services/teacher_training_public_api/sync_provider_spec.rb

```diff
@@ -6,42 +6,531 @@
+          specified_attributes: [{
+            code: 'ABC1',
+            accredited_body_code: nil,
+            state: 'withdrawn',
+          }],
+        )
+        stub_teacher_training_api_sites(
+          provider_code: 'ABC',
+          course_code: 'ABC1',
+          specified_attributes: [{
+            code: 'A',
+          }],
+        )
+
+        described_class.new(provider_from_api: provider_from_api, recruitment_cycle_year: stubbed_recruitment_cycle_year).call
+        expect(CourseOption.count).to eq 1
+        Course.first.update!(withdrawn: false)
+
+        described_class.new(provider_from_api: provider_from_api, recruitment_cycle_year: stubbed_recruitment_cycle_year).call
+        expect(Course.first.withdrawn).to eq true
+      end
+
+      it 'sets the accredited provider' do
+        create :provider, code: 'DEF', name: 'Foobar College'
+
+        provider_from_api = fake_api_provider(code: 'ABC')
+        stub_teacher_training_api_courses(
+          provider_code: 'ABC',
+          specified_attributes: [{
+            code: 'ABC1',
+            accredited_body_code: 'DEF',
+          }],
+        )
+        stub_teacher_training_api_sites(
+          provider_code: 'ABC',
+          course_code: 'ABC1',
+          specified_attributes: [{
+            code: 'A',
+          }],
+        )
+
+        described_class.new(provider_from_api: provider_from_api, recruitment_cycle_year: stubbed_recruitment_cycle_year).call
+        course_option = CourseOption.first
+        expect(course_option.course.accredited_provider.code).to eq 'DEF'
+        expect(course_option.course.accredited_provider.name).to eq 'Foobar College'
+      end
+
+      it 'does not set the accredited provider if it is the same as the training provider' do
+        provider_from_api = fake_api_provider(code: 'ABC')
+        stub_teacher_training_api_courses(
+          provider_code: 'ABC',
+          specified_attributes: [{
+            code: 'ABC1',
+            accredited_body_code: 'ABC',
+          }],
+        )
+        stub_teacher_training_api_sites(
+          provider_code: 'ABC',
+          course_code: 'ABC1',
+          specified_attributes: [{
+            code: 'A',
+          }],
+        )
+
+        described_class.new(provider_from_api: provider_from_api, recruitment_cycle_year: stubbed_recruitment_cycle_year).call
+        expect(Course.find_by(code: 'ABC1').accredited_provider).to be_nil
+      end
+
+      it 'resets the accredited provider if it is no longer specified' do
+        course = create(:course, accredited_provider: create(:provider), code: 'ABC1', provider: create(:provider, code: 'ABC'))
+
+        provider_from_api = fake_api_provider(code: 'ABC')
+        stub_teacher_training_api_courses(
+          provider_code: 'ABC',
+          specified_attributes: [{
+            code: 'ABC1',
+            accredited_body_code: nil,
+          }],
+        )
+        stub_teacher_training_api_sites(
```

Comment:
This is here because we always make a call to get the sites for a course. Since it is not relevant to the test, I'll look at making a helper method to tidy this up. Hopefully making the tests less verbose and more focused on what is being tested.

Reply:
Done